.github/workflows/build.yaml

```diff
@@ -103,12 +103,13 @@ jobs:
           if-no-files-found: warn
           retention-days: 7
       - name: 'Run UI integration tests'
+        id: run_tests
         working-directory: ./liberty-tools-intellij
         run: bash ./src/test/resources/ci/scripts/run.sh
       - name: 'Archive Test logs and reports'
-        if: ${{ failure() }}
-        uses: actions/upload-artifact@v3
+        if: ${{ failure() && steps.run_tests.conclusion == 'failure' }}
+        uses: actions/upload-artifact@v4.3.4
         with:
-          name: ${{ matrix.reportName }}
+          name: ${{ matrix.reportName }}-LTI-${{ env.REF_LTI_TAG != '' && env.REF_LTI_TAG || 'default' }}-LSP4IJ-${{ env.LSP4IJ_BRANCH }}
```

Comment:
Is  `env.REF_LTI_TAG != ''` needed? By default || operator handles the case where env.REF_LTI_TAG might be empty or undefined by falling back to 'default'.

Reply:
Yes @anusreelakshmi934. But this will take effect once this PR https://github.com/OpenLiberty/liberty-tools-intellij/pull/864 is merged. That PR is having `env.REF_LTI_TAG`  defined. I have tested with the changes and it is working fine as espected and tested that there is no issue with merging this PR first or merging the above-mentioned PR first.

Reply:
REF_LTI_TAG input is not defined in this workflow file.

Reply:
Ok, as mentioned in our Kochi team scrum, I am making this PR a `draft`, since it has a dependency (env.REF_LTI_TAG) on another PR, https://github.com/OpenLiberty/liberty-tools-intellij/pull/864, which is already open. After that PR is merged, I will test this PR again and make it `ready for review`.

Reply:
I remember just reviewing a similar change in the PR you mentioned. Can I assume you retested after merging?

Reply:
Yes @mrglavas , I have retested again with changes after merged the PR 864

Reply:
Thanks @vaisakhkannan. Wanted to confirm that before I approved. I've approved the PR now.